contracts/RelayHub.sol

```diff
@@ -114,24 +117,26 @@ contract RelayHub is IRelayHub {
         emit Withdrawn(account, dest, amount);
     }
 
-    function verifyGasLimits(
+    function verifyGasAndDataLimits(
         uint256 paymasterMaxAcceptanceBudget,
         GsnTypes.RelayRequest calldata relayRequest,
         uint256 initialGas
     )
     private
     view
-    returns (IPaymaster.GasLimits memory gasLimits, uint256 maxPossibleGas) {
-        gasLimits =
-            IPaymaster(relayRequest.relayData.paymaster).getGasLimits{gas:50000}();
-
-        require(paymasterMaxAcceptanceBudget >= gasLimits.acceptanceBudget, "unexpected high acceptanceBudget");
+    returns (IPaymaster.GasAndDataLimits memory gasAndDataLimits, uint256 maxPossibleGas) {
+        gasAndDataLimits =
+            IPaymaster(relayRequest.relayData.paymaster).getGasAndDataLimits{gas:50000}();
+        require(msg.data.length <= gasAndDataLimits.calldataSizeLimit, "msg.data exceeded limit" );
```

Comment:
If the next version of a relay hub uses x2 the `_dataGasCostPerByte`, this Paymaster will not be able to update to use it.

Reply:
This only validates paymasters' own requirement of data length, how does it change through hubs?